src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameLexer.cs

```diff
@@ -0,0 +1,392 @@
+        /// </summary>
+        public StackFrameToken? TryScanPath()
+        {
+            var inTrivia = TryScanInTrivia();
+            if (!inTrivia.HasValue)
+            {
+                return null;
+            }
+
+            var startPosition = Position;
+
+            while (Position < Text.Length)
+            {
+                // Path needs to do a look ahead to determine if adding the next character
+                // invalidates the path. Break if it does.
+                //
+                // This helps to keep the complex rules for what FileInfo does to validate path by calling to it directly.
+                // We can't simply check all invalid characters for a path because location in the path is important, and we're not 
+                // in the business of validating if something is correctly a file. It's cheap enough to let the FileInfo constructor do that. The downside 
+                // is that we are constructing a new object with every pass. If this becomes problematic we can revisit and fine a more 
+                // optimized pattern to handle all of the edge cases.
+                //
+                // Example: C:\my\path \:other
+                //                      ^-- the ":" breaks the path, but we can't simply break on all ":" (which is included in the invalid characters for Path)
+                //
+
+                var str = GetSubSequence(startPosition, Position + 1).CreateString();
+
+                var isValidPath = IOUtilities.PerformIO(() =>
+                {
+                    var fileInfo = new FileInfo(str);
+                    return true;
+                }, false);
+
+                if (!isValidPath)
+                {
+                    break;
+                }
+
+                Position++;
+            }
+
+            if (startPosition == Position)
+            {
+                return CreateToken(StackFrameKind.InvalidPathToken, ImmutableArray.Create(inTrivia.Value), VirtualCharSequence.Empty);
+            }
+
+            return CreateToken(StackFrameKind.PathToken, ImmutableArray.Create(inTrivia.Value), GetSubSequenceToCurrentPos(startPosition));
+        }
+
+        /// <summary>
+        /// Returns a number token with the <see cref="StackFrameKind.LineTrivia"/> and remainging <see cref="StackFrameKind.SkippedTextTrivia"/>
+        /// attached to it. If no numbers are found, returns a <see cref="StackFrameKind.None"/> token with the trivia that was scanned.
+        /// </summary>
+        /// <returns></returns>
+        public StackFrameToken? TryScanLineNumber()
+        {
+            var lineTrivia = TryScanLineTrivia();
+            if (!lineTrivia.HasValue)
+            {
+                return null;
+            }
+
+            var numberToken = TryScanNumbers();
+            if (!numberToken.HasValue)
+            {
+                return CreateToken(StackFrameKind.None, ImmutableArray.Create(lineTrivia.Value), VirtualCharSequence.Empty);
```

Comment:
this is generally a no-no.  you can't have a None token with trivia on it.  None is only for the totally absent/empty token.

Reply:
Added `InvalidNumberToken` and changed to that